src/lib/utils/os_utils.cpp

```diff
@@ -55,6 +55,12 @@
   extern "C" char **environ;
 #endif
 
+#if defined(BOTAN_TARGET_OS_IS_IOS) || defined(BOTAN_TARGET_OS_IS_MACOS)
+  #define LOCKED_PAGES_FD (255<<24)
+#else
+  #define LOCKED_PAGES_FD -1
```

Comment:
Use `const int locked_fd = ...` instead of a define here, and add a comment explaining meaning of/use for the Apple value. LGTM otherwise